alica_engine/src/engine/collections/Variant.cpp

```diff
@@ -3,7 +3,43 @@
 namespace alica
 {
 
-constexpr size_t Variant::kUnionSize;
-constexpr size_t Variant::kVariantSize;
+void variant::serializeTo(std::string& arr, const Variant& var)
+{
+    arr.clear();
+    arr += std::to_string(var.index()); // type
+
+    std::visit( [&](auto&& arg) {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (!(std::is_same_v<std::monostate, T>)) {
+            arr += (std::to_string(arg));
+        }
+    }, var );
+}
+
+void variant::loadFrom(const std::string& arr, Variant& var)
+{
+    if (arr.size() < 2) { // should atleast contain once char for type and one char for the data
```

Comment:
Maybe this should throw exception?

Reply:
i think it's reasonable to return an empty variant if data provided is empty